py/src/braintrust/logger.py

```diff
@@ -1121,6 +1147,65 @@ def __exit__(self, type, value, callback):
         self.end()
 
 
+class _ObjectLoader:
+    def __init__(self, object_type: str, object_id: id, version: "str | int" = None):
+        self.finished = False
+        self.object_type = object_type
+        self.object_id = object_id
+
+        self._fetched_data = None
+
+        self._pinned_version = None
+        if version is not None:
+            try:
+                self._pinned_version = int(version)
+                assert self._pinned_version >= 0
+            except (ValueError, AssertionError):
+                raise ValueError(f"version ({version}) must be a positive integer")
+
+    def fetch(self):
+        self._check_not_finished()
+
+        for record in self.fetched_data:
+            yield {
+                "id": record.get("id"),
+                "input": json.loads(record.get("input") or "null"),
```

Comment:
Why not `None` instead of a "null" string? Also the `or` part is a bit scary, because it will apply to any "zero-valued" python object. If we only expect the value of `record.get("input")` to be None or a non-zero-valued python object (I don't think empty string is valid json?), then it's fine, but at that point, may as well write `record.get("input", "null")` (or just leave it as `None`?).